packages/react/src/lavadome.jsx

```diff
@@ -1,26 +1,34 @@
 import React, { useEffect, useRef } from 'react'
 import { LavaDome as LavaDomeCore } from "@lavamoat/lavadome-core"
-import {create, hasOwn, stringify, WeakMap, get, set} from "@lavamoat/lavadome-core/src/native.mjs";
+import {create, hasOwn, WeakMap, get, set, isArray, at, Error} from "@lavamoat/lavadome-core/src/native.mjs";
 
 const
+    tokenToCopyInvokerMap = new WeakMap(),
     tokenToTextMap = new WeakMap(),
     tokenToDepMap = new WeakMap(),
     textToTokenMap = create(null);
 
-// map sensitive text of the user with a unique token representing it, so that the
-// token is the one being passed around React internals rather than the sensitive text
-export const textToToken = text => {
-    if (typeof text !== 'string') {
-        throw new Error(`LavaDomeReact: first argument must be a string, instead got ${stringify(text)}`);
+// transform provided sensitive text into lavadome capabilities such as:
+// 1. unique token representing the text, so that it's the one tossed around React internals instead of the sensitive text
+// 2. copy callback that's when invoked copies the sensitive text to clipboard
+export const textToLavaDomeCapabilities = text => {
+    const type = typeof text;
+    if (type !== 'string') {
+        throw new Error(
+            `LavaDomeReact: first argument must be a string, instead got ${type}`);
     }
 
     if (!hasOwn(textToTokenMap, text)) {
         const token = create(null);
         textToTokenMap[text] = token;
         set(tokenToTextMap, token, text);
+        set(tokenToCopyInvokerMap, token, () => {});
     }
 
-    return textToTokenMap[text];
+    const token = textToTokenMap[text];
+    const copy = () => get(tokenToCopyInvokerMap, token)();
```

Comment:
should this be a bound function or does it not matter?

Reply:
the copy function stores what it needs within the closure that was originally constructed for it, which is the alternative to bounding in this case